skimage/morphology/_flood_fill.py

```diff
@@ -0,0 +1,274 @@
+"""flood_fill.py - inplace flood fill algorithm
+
+This module provides a function to fill all equal (or within tolerance) values
+connected to a given seed point with a different value.
+"""
+
+import numpy as np
+import warnings
+
+from .extrema import (_resolve_neighborhood, _set_edge_values_inplace,
+                      _fast_pad)
+from .watershed import _offsets_to_raveled_neighbors
+from ._flood_fill_cy import _flood_fill_equal, _flood_fill_tolerance
+
+
+def flood_fill(image, seed_point, new_value, *, selem=None, connectivity=None,
+               tolerance=None, inplace=False):
+    """Perform flood filling on an image.
+
+    Starting at a specific `seed_point`, connected points equal or within
+    `tolerance` of the seed value are found, then set to `new_value`.
+
+    Parameters
+    ----------
+    image : ndarray
+        An n-dimensional array.
+    seed_point : tuple or int
+        The point in `image` used as the starting point for the flood fill.  If
+        the image is 1D, this point may be given as an integer.
+    new_value : `image` type
+        New value to set the entire fill.  This must be chosen in agreement
+        with the dtype of `image`.
+    selem : ndarray, optional
+        A structuring element used to determine the neighborhood of each
+        evaluated pixel. It must contain only 1's and 0's, have the same number
+        of dimensions as `image`. If not given, all adjacent pixels are
+        considered as part of the neighborhood (fully connected).
+    connectivity : int, optional
+        A number used to determine the neighborhood of each evaluated pixel.
+        Adjacent pixels whose squared distance from the center is larger or
```

Comment:
This description doesn't seem right.

Reply:
This is literally identical to how we describe connectivity elsewhere, most notably https://github.com/scikit-image/scikit-image/blob/29a22e17c8b4ccf26383e3007ab014eb18b81f12/skimage/morphology/extrema.py#L351

Reply:
It may be correct/incorrect in other places as well; but can we check carefully that this formulation is accurate?  It may well be, but I wanted more eyes on it.

Reply:
The word `larger` here is particularly suspect.

```
In [8]: connectivity = 1; ndi.generate_binary_structure(2, connectivity)                                                                                                             
Out[8]: 
array([[False,  True, False],
       [ True,  True,  True],
       [False,  True, False]])
```

Note the description from `ndimage`:

```
connectivity : int
     `connectivity` determines which elements of the output array belong
     to the structure, i.e. are considered as neighbors of the central
     element. Elements up to a squared distance of `connectivity` from
     the center are considered neighbors. `connectivity` may range from 1
     (no diagonal elements are neighbors) to `rank` (all elements are
     neighbors).
```

Reply:
Oh, I think I messed that up when writing the docstring for `local_maxima`. I even remember looking up that docstring as a source. :see_no_evil:  If we change "larger" into "smaller" it should be correct though.